dvc/compare.py

```diff
@@ -37,11 +38,24 @@ def __init__(self, columns: Sequence[str], fill_value: str = ""):
         self._columns: Dict[str, Column] = {name: Column() for name in columns}
         self._keys: List[str] = list(columns)
         self._fill_value = fill_value
+        self._protected: Set[str] = set()
 
     @property
     def columns(self) -> List[Column]:
         return list(map(self.column, self.keys()))
 
+    @property
+    def protected(self) -> Set[str]:
```

Comment:
For myself, I would use an `is_protected` function here and make the `re.match` logic inside it.  Decouple them from outside.